ggshield/pypi.py

```diff
@@ -0,0 +1,107 @@
+import re
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Dict, List, Set
+
+import click
+
+from ggshield.config import Config
+from ggshield.output import OutputHandler
+from ggshield.path import get_files_from_paths
+from ggshield.scan import Files, Result, ScanCollection
+from ggshield.utils import SupportedScanMode
+
+
+PYPI_DOWNLOAD_TIMEOUT = 30
+
+
+def save_package_to_tmp(temp_dir: str, package_name: str) -> None:
+    command: List[str] = [
+        "pip",
+        "download",
+        package_name,
+        "--dest",
+        temp_dir,
+        "--no-deps",
+    ]
+
+    try:
+        click.echo("Downloading pip package... ", nl=False)
+        subprocess.run(
+            command,
+            check=True,
+            stderr=subprocess.PIPE,
+            timeout=PYPI_DOWNLOAD_TIMEOUT,
+        )
+        click.echo("OK")
+
+    except subprocess.CalledProcessError:
+        raise click.ClickException(f'Failed to download "{package_name}"')
+
+    except subprocess.TimeoutExpired:
+        raise click.ClickException('Command "{}" timed out'.format(" ".join(command)))
+
+
+def get_files_from_package(
+    temp_dir: str,
+    package_name: str,
+    exclusion_regexes: Set[re.Pattern],
+    verbose: bool,
+) -> Files:
+    temp_dir_path = Path(temp_dir)
+    archive: Path = next(temp_dir_path.iterdir())
+    unpack_kwargs: Dict[str, str] = (
+        {"format": "zip"} if archive.suffix == ".whl" else {}
+    )
+
+    try:
+        shutil.unpack_archive(str(archive), extract_dir=temp_dir_path, **unpack_kwargs)
+    except Exception as exn:
+        raise click.ClickException(f'Failed to unpack package "{package_name}": {exn}.')
+
+    exclusion_regexes.add(re.compile(re.escape(archive.name)))
+
+    return get_files_from_paths(
+        paths=[temp_dir],
+        exclusion_regexes=exclusion_regexes,
+        recursive=True,
+        yes=True,
+        verbose=verbose,
+        ignore_git=True,
+    )
+
+
+@click.command()
+@click.argument("package_name", nargs=1, type=click.STRING, required=True)
+@click.pass_context
+def pypi_cmd(ctx: click.Context, package_name: str) -> int:  # pragma: no cover
+    """
+    scan a pypi package <NAME>.
+    """
+    config: Config = ctx.obj["config"]
+    output_handler: OutputHandler = ctx.obj["output_handler"]
+
+    with tempfile.TemporaryDirectory(suffix="ggshield") as temp_dir:
+        save_package_to_tmp(temp_dir=temp_dir, package_name=package_name)
+
+        files: Files = get_files_from_package(
```

Comment:
`get_files_from_package ` already type-hints the return type, so I don't see why type-hint the variable here

Reply:
It seems to me that it becomes too complex to know whether we should type hint or not. Can type hint a variable be a problem?